src/fidesops/task/graph_task.py

```diff
@@ -486,6 +500,26 @@ def collect_queries_fn(
     return env
 
 
+def update_mapping_from_cache(
+    dsk: Dict[CollectionAddress, Tuple[Any, ...]],
+    resources: TaskResources,
+    start_fn: Callable,
+) -> None:
+    """When resuming a privacy request from a paused or failed state, update the `dsk` dictionary with results we've
+    already obtained from a previous run. Remove upstream dependencies for these nodes, and just return the data we've
+    already retrieved, rather than visiting them again.
+
+    If there's no cached data, the dsk dictionary won't change.
+    """
+
+    cached_results: Dict[str, Optional[List[Row]]] = resources.get_all_cached_objects()
+
+    for collection_name in cached_results:
+        dsk[CollectionAddress.from_string(collection_name)] = (
+            start_fn(cached_results[collection_name]),
+        )
+
```

Comment:
This is a primary change.  This will update the graph when we attempt to re-run - it will still have the same number of collections as before, we just may not need to run actual queries on certain nodes, we'll use the data we already have.